docs(Cupertino): Add documentation for Uno.Cupertino (#6185)
unoplatform/uno: doc/articles/guides/uno-cupertino-walkthrough.md

# How to use Uno.Cupertino

> This guide will walk you through the necessary steps to setup and to use the [`Uno.Cupertino` package](https://www.nuget.org/packages/Uno.Cupertino) in an Uno Platform application.

Reviewer: ```suggestion
This guide will walk you through the necessary steps to set up and use the [`Uno.Cupertino` package](https://www.nuget.org/packages/Uno.Cupertino) in an Uno Platform application.
```

* [**Mono**](https://www.mono-project.com/download/stable/)

* **.NET Core SDK**
    * [.NET Core 3.1 SDK](https://dotnet.microsoft.com/download/dotnet-core/3.1) (**version 3.1.8 (SDK 3.1.402)** or later)
    * [.NET Core 5.0 SDK](https://dotnet.microsoft.com/download/dotnet-core/5.0) (**version 5.0 (SDK 5.0.100)** or later)

    > Use `dotnet --version` from the terminal to get the version installed.

# [Visual Studio for Mac](#tab/tabid-vsmac)

* [**Visual Studio for Mac 8.8**](https://visualstudio.microsoft.com/vs/mac/)
* [**Xcode**](https://apps.apple.com/us/app/xcode/id497799835?mt=12) 10.0 or higher
* An [**Apple ID**](https://support.apple.com/en-us/HT204316)
* **.NET Core SDK**
    * [.NET Core 3.1 SDK](https://dotnet.microsoft.com/download/dotnet-core/3.1) (**version 3.1.8 (SDK 3.1.402)** or later)
    * [.NET Core 5.0 SDK](https://dotnet.microsoft.com/download/dotnet-core/5.0) (**version 5.0 (SDK 5.0.100)** or later)
* [**GTK+3**](https://formulae.brew.sh/formula/gtk+3) for running the Skia/GTK projects

# [JetBrains Rider](#tab/tabid-rider)

* [**Rider Version 2020.2+**](https://www.jetbrains.com/rider/download/)

> * [**Rider Xamarin Android Support Plugin**](https://plugins.jetbrains.com/plugin/12056-rider-xamarin-android-support/) (you may install it directly from Rider)

Reviewer 2: We should redirect to the existing documentation to avoid duplication (this changes frequently)

    Here are some issues that you are likely to run into after complete the above step:
    - > NU1605: Detected package downgrade: Uno.UI from 3.6.0-dev.275 to 3.5.1. Reference the package directly from the project to select a different version.

        The app may not compile, crash at runtime, or behave strangely as a result of this.
        solution: You need to update the version of `Uno.UI` packages for all project heads that you are using to the higher version.
        > note: By `Uno.UI` packages, it includes:
        > - Uno.UI
        > - Uno.UI.RemoteControl
        > - Uno.UI.WebAssembly
        > - Uno.UI.Skia.Gtk
        > - Uno.UI.Skia.Tizen
        > - Uno.UI.Skia.Wpf

    - When building the `.Droid` project, the project failed to build with:
        ```
        error : Could not find 1 Android X assemblies, make sure to install the following NuGet packages:
            - Xamarin.AndroidX.Lifecycle.LiveData
        You can also copy-and-paste the following snippet into your .csproj file:
            <PackageReference Include="Xamarin.AndroidX.Lifecycle.LiveData" Version="2.1.0" />
        ```

>         solution: Simply add the specific version of `Xamarin.AndroidX.Lifecycle.LiveData` to the `.Droid` project

Reviewer 2: ```suggestion
        solution: Add the mentioned version of `Xamarin.AndroidX.Lifecycle.LiveData` to the `.Droid` project
```

1. Add the following code inside `App.xaml`:
    ```xml
    <Application.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <!-- Load WinUI resources -->
                <XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls" />

                <CupertinoColors xmlns="using:Uno.Cupertino"  />

> 			 	<CupertinoResources xmlns="using:Uno.Cupertino" />

Reviewer 3: spacing
```suggestion
                <CupertinoResources xmlns="using:Uno.Cupertino" />
```

> 				<CupertinoResources xmlns="using:Uno.Cupertino" />

Reviewer 2: ```suggestion
			 	<CupertinoResources xmlns="using:Uno.Cupertino" />
```

                <!-- Application's custom styles -->
                <!-- other ResourceDictionaries -->
            </ResourceDictionary.MergedDictionaries>
        </ResourceDictionary>
    </Application.Resources>
    ```

### Section 2: Using Uno.Cupertino library
1. Let's add a few controls with the Cupertino style to `MainPage.xaml`:
    ```xml
    <Page x:Class="UnoCupertinoSample.MainPage"
          xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
          xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
          xmlns:toolkit="using:Uno.UI.Toolkit">

        <Grid toolkit:VisibleBoundsPadding.PaddingMask="Top" >
            <ScrollViewer>
                <StackPanel Margin="16,0" Spacing="8">

>                     <!-- controls with cupertino styles -->

Reviewer 2: ```suggestion
                    <!-- controls with Cupertino styles -->
```

                    <Button Content="button" Style="{StaticResource CupertinoButtonStyle}" />
                    <ComboBox ItemsSource="asd" Style="{StaticResource CupertinoComboBoxStyle}" />
                    <DatePicker Style="{StaticResource CupertinoDatePickerStyle}" />
                    <TextBox Text="input" Style="{StaticResource CupertinoTextBoxStyle}" />

                </StackPanel>
            </ScrollViewer>
        </Grid>
    <Page>
    ```

> [TIP!]
> You can find the style names using these methods:
> - "Feature" section of Uno.Themes README: https://github.com/unoplatform/Uno.Themes#features
> - Going through the source code of control styles: https://github.com/unoplatform/Uno.Themes/tree/master/src/library/Uno.Cupertino/Styles/Controls
> - Check out the [Uno.Gallery web app](https://gallery.platform.uno/) (Click on the `<>` button to view xaml source)

### Section 3: Overriding Color Palette
1. Create the nested folders `Styles\` and then `Styles\Application\` under the `.Shared` project
1. Add a new Resource Dictionary `ColorPaletteOverride.xaml` under `Styles\Application\`

> 1. Replace the content of that res-dict with the source from: https://github.com/unoplatform/Uno.Themes/blob/master/src/library/Uno.Cupertino/Styles/Application/ColorPalette.xaml

Reviewer: ```suggestion
1. Replace the content of that `ResourceDictionary` with the source from: https://github.com/unoplatform/Uno.Themes/blob/master/src/library/Uno.Cupertino/Styles/Application/ColorPalette.xaml
```

1. Make a few changes to the colors:
    > Here we are replacing the last 2 characters with 00, essentially dropping the blue-channel
    ```xml
    <!-- Light Theme -->
    <ResourceDictionary x:Key="Light">
        <Color x:Key="CupertinoBlueColor">#007B00</Color>
        <!-- ... -->
	</ResourceDictionary>

    <!-- Dark Theme -->
    <ResourceDictionary x:Key="Dark">
        <Color x:Key="CupertinoBlueColor">#007B00</Color>
        <!-- ... -->
    </ResourceDictionary>

    <!-- ... -->
    ```

1. In `App.xaml`, update the line that initializes the `CupertinoColors` to include the new palette override:    
    ```diff
    <Application.Resources>
		<ResourceDictionary>
			<ResourceDictionary.MergedDictionaries>
				<!-- ... -->

	-			<CupertinoColors xmlns="using:Uno.Cupertino" />
	+			<CupertinoColors xmlns="using:Uno.Cupertino" 

>     +                            ColorPaletteOverrideSource=ms-appx:///Styles/Application/ColorPaletteOverride.xaml" />

Reviewer 2: ```suggestion
   +                            ColorPaletteOverrideSource=ms-appx:///Styles/Application/ColorPaletteOverride.xaml" />
```
